conv/convert_format_test.go

```diff
@@ -0,0 +1,344 @@
+			require.Error(t, err, testErrMsg(f))
+
+			_, err = ConvertInteger[int64](f)
+			require.Error(t, err, testErrMsg(f))
+		}
+	})
+}
+
+func TestConvertUinteger(t *testing.T) {
+	t.Run("with uint8", func(t *testing.T) {
+		validInts := []uint8{0, 1, math.MaxUint8}
+		invalidInts := []string{"1.233", "a", "false", strconv.FormatUint(math.MaxUint64, 10), strconv.FormatInt(-1, 10)}
+
+		for _, f := range validInts {
+			str := FormatUinteger(f)
+			c1, err := ConvertUint8(str)
+			require.NoError(t, err)
+			assert.EqualValues(t, f, c1)
+
+			c2, err := ConvertUinteger[uint8](str)
+			require.NoError(t, err)
+			assert.EqualValues(t, c1, c2)
+		}
+
+		for _, f := range invalidInts {
+			_, err := ConvertUint8(f)
+			require.Error(t, err, testErrMsg(f))
+
+			_, err = ConvertUinteger[uint8](f)
+			require.Error(t, err, testErrMsg(f))
+		}
+	})
+
+	t.Run("with uint16", func(t *testing.T) {
+		validUints := []uint16{0, 1, math.MaxUint8, math.MaxUint16}
+		invalidUints := []string{"1.233", "a", "false", strconv.FormatUint(math.MaxUint64, 10), strconv.FormatInt(-1, 10)}
+
+		for _, f := range validUints {
+			str := FormatUinteger(f)
+			c1, err := ConvertUint16(str)
+			require.NoError(t, err)
+			assert.EqualValues(t, f, c1)
+
+			c2, err := ConvertUinteger[uint16](str)
+			require.NoError(t, err)
+			assert.EqualValues(t, c1, c2)
+		}
+
+		for _, f := range invalidUints {
+			_, err := ConvertUint16(f)
+			require.Error(t, err, testErrMsg(f))
+
+			_, err = ConvertUinteger[uint16](f)
+			require.Error(t, err, testErrMsg(f))
+		}
+	})
+
+	t.Run("with uint32", func(t *testing.T) {
+		validUints := []uint32{0, 1, math.MaxUint8, math.MaxUint16, math.MaxUint32}
+		invalidUints := []string{"1.233", "a", "false", strconv.FormatUint(math.MaxUint64, 10), strconv.FormatInt(-1, 10)}
+
+		for _, f := range validUints {
+			str := FormatUinteger(f)
+			c1, err := ConvertUint32(str)
+			require.NoError(t, err)
+			assert.EqualValues(t, f, c1)
+
+			c2, err := ConvertUint32(str)
+			require.NoError(t, err)
+			assert.EqualValues(t, c1, c2)
+		}
+
+		for _, f := range invalidUints {
+			_, err := ConvertUint32(f)
+			require.Error(t, err, testErrMsg(f))
+
+			_, err = ConvertUinteger[uint32](f)
+			require.Error(t, err, testErrMsg(f))
+		}
+	})
+
+	t.Run("with uint64", func(t *testing.T) {
+		validUints := []uint64{0, 1, math.MaxUint8, math.MaxUint16, math.MaxUint32, math.MaxUint64}
+		invalidUints := []string{"1.233", "a", "false", strconv.FormatInt(-1, 10), uint64OverflowStr()}
+
+		for _, f := range validUints {
+			str := FormatUinteger(f)
+			c1, err := ConvertUint64(str)
+			require.NoError(t, err)
+			assert.EqualValues(t, f, c1)
+
+			c2, err := ConvertUinteger[uint64](str)
+			require.NoError(t, err)
+			assert.EqualValues(t, c1, c2)
+		}
+		for _, f := range invalidUints {
+			_, err := ConvertUint64(f)
+			require.Error(t, err, testErrMsg(f))
+
+			_, err = ConvertUinteger[uint64](f)
+			require.Error(t, err, testErrMsg(f))
+		}
+	})
+}
```

Comment:
It's interesting how you are coding/testing I made on my own lib 

https://github.com/ccoVeille/go-safecast

That is about converting things and check if they overflow

It may inspire me for refactoring mine, or to add tests

I like the usage of math/big to create a string that overflows uint64

Reply:
I created myself a reminder 

- https://github.com/ccoVeille/go-safecast/issues/99

Reply:
Btw this reminds me that we have an old stuff pending on the validation side, when json numbers overflow float64...